electron/app/components/charts/BarChart.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect } from 'react';
+import { Bar } from 'react-chartjs-2';
+import * as sqlite3 from 'sqlite3';
+import log from 'electron-log';
+import interpolateColors from '../../utils/colors';
+
+interface BarChartProps {
+  db: sqlite3.Database;
+  chartQuery: () => Promise<WordCountTypes.Results | TopFriendsTypes.Results>;
+  titleText: string;
+  subLabel: string;
+  /* TODO:
+   * The *Axiskeys can only be
+   * WordCountTypes.Result | TopFriendsTypes.Results
+   */
+  xAxisKey: string;
+  yAxisKey: string;
+  stacked?: boolean;
+  colorInterpolationFunc: (t: number) => string;
+}
+
+export default function BarChat(props: BarChartProps) {
+  const {
+    chartQuery,
+    titleText,
+    subLabel,
+    xAxisKey,
+    yAxisKey,
+    colorInterpolationFunc,
+  } = props;
+  const [xAxisData, setXAxisData] = useState<string[]>([]);
+  const [yAxisData, setYAxisData] = useState<number[]>([]);
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const dataList = await chartQuery();
+        setXAxisData(
+          (dataList as Array<
+            WordCountTypes.ChartData | TopFriendsTypes.ChartData
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          >).map((obj: { [x: string]: any }) => obj[xAxisKey])
+        );
+        setYAxisData(
+          (dataList as Array<
+            WordCountTypes.ChartData | TopFriendsTypes.ChartData
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          >).map((obj: { [x: string]: any }) => obj[yAxisKey])
+        );
```

Comment:
This is hideous, but I don't know how to get .map in Typescript land to not complain about union-ed types

Also the `{ [x: string]: any }` is annoying. This would be fixed if the TODO here is fixed. https://github.com/Left-on-Read/app/pull/23/files#diff-708c132e2c311a4f66441bcce712052627a8206180cce39b11fbcf438d32cb84R12-R15 

Anyone know how to do so?